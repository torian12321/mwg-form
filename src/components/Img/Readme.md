We require the use of src and alt, only enforced by react in dev mode

#### Basic
Use this just for wireframing
```
<Img alt="desc" />
```

#### Without Alt attribute
If there is no alt, the picture will be considered as presentation
```
<Img />
```

#### Custom src
Use a local image or from a url
```
const imgExample = require ('./example.jpg');

<div>
  <Img alt="img 1" src='http://placeimg.com/200/150/nature' />
  <Img alt="img 2" src={imgExample} />
</div>
```
