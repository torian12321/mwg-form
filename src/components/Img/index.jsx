import React      from 'react';
import PropTypes  from 'prop-types';
import ImgDef     from '../../img/default/img.png';

class Img extends React.PureComponent {
  render() {
    const { src, alt, ...other } = this.props;

    return (
      <img
        {...other}
        src ={src}
        alt ={alt}
        role={!alt && 'presentation'}
      />
    )
  }
}

Img.propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string
};
Img.defaultProps = {
  src: ImgDef
};

module.exports = Img;
