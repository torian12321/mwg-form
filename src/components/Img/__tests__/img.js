import React    from 'react';
import renderer from 'react-test-renderer';
import Img      from '../index';

describe('component', () => {
  describe('<Img />', () => {

    it('should render correctly', () => {
      expect(renderer.create(
        <Img src='img.png' />
      )).toMatchSnapshot();
    });

    it('should render with alt text', () => {
      expect(renderer.create(
        <Img src='img.png' alt='image description' />
      )).toMatchSnapshot();
    });

  });
});