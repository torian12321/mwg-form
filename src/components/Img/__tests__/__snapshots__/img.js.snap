// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <Img /> should render correctly 1`] = `
<img
  alt={undefined}
  role="presentation"
  src="img.png"
/>
`;

exports[`component <Img /> should render with alt text 1`] = `
<img
  alt="image description"
  role={false}
  src="img.png"
/>
`;
